### ISSUE TITLE
add ability to ignore certain events

### ISSUE THREAD

**Issue by danmarsden** [catalyst/moodle-block_dedication#64]
Certain events in the logs shouldn't really be used for caclulating time-spent.

we have a custom event on one site that is triggered when the session is about to expire - that's a good one to target for now.

**Comment by reskit** [comment]
Related to this issue we've found in one of our sites a course which had more than 3 milions records generated by cron (user admin) and the event who generated those records was _\core\event\question_deleted_.

The task _\block_dedication\task\dedication_collector_ crashed every time with memory exhausted error.

I've solved this issue filtering all events generated by its origin (cli) because the logs generated by cron are not useful for calculation time-spent. The solution implies modify the SQL inside get_user_dedication function.

If you think it's a valid solution, I could make a pull request.

Thanks for maintaining this plugin, it's really useful for us.

**Comment by danmarsden** [comment]
yeah good call - makes sense to ignore CLI based events completely and they shouldn't contribute to time-spent in any way. If you have internal development capabilities feel free to submit a pull request otherwise we'll probably make this change at some point.

thanks!

**Comment by danmarsden** [comment]
merged in a change to filter out CLI events - thanks @reskit
